### PR TITLE
[improvement]avoid read bitmap index when no index works

### LINKED ISSUE #8388: [Enhancement] Avoid read index bitmap when no index is provided
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

When a query hits index, then index bitmap is reset;
Storage layer read rows by index bitmap.
In the current implementation, even a query doesn't hit any index, storage still need to read rows by index bitmap.
This could bring two costs:
1 Read data from roaring bitmap.
2 The foreach to read rowids from roaring bitmap.

There is still optimization room for query without indexes.

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wangbo**
# Proposed changes

Issue Number: close #8388

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

Performance test has been added, see #8388

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/segment_iterator.h [hunk 1]
```diff
@@ -45,6 +45,11 @@ class BitmapIndexIterator;
 class BitmapIndexReader;
 class ColumnIterator;
 
+enum ReadIndexType {
+    READ_WITH_BITMAP,
+    READ_WITHOUT_BITMAP,
```
Comment (zbtzbtzbt) [inline be/src/olap/rowset/segment_v2/segment_iterator.h:50]: we can remove the end `,`

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 2]
```diff
@@ -76,6 +85,27 @@ class SegmentIterator::BitmapRangeIterator {
         return true;
     }
 
+    bool get_next_range_without_bitmap_index(uint32_t max_range_size, uint32_t* from, uint32_t* to) {
+        if (_curr_pos == _max_value) {
+            return false;
+        }
+        if (_curr_pos + max_range_size <= _max_value) {
```
Comment (zbtzbtzbt) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:92]: should we add `_eof` like `get_next_range_with_bitmap_index()`? ( i am not sure)

Comment (wangbo) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:92]: ```_eof``` is used to confirm whether reachs the end of ```roaring_read_uint32_iterator```.
If you don't use ```roaring_read_uint32_iterator```, then ```_eof``` is useless.

Comment (zbtzbtzbt) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:92]: i get it

**Pull request closed without merging**